### ISSUE TITLE
High CPU utilization even when idle starting with 1.17.0

### ISSUE THREAD

**Issue by jflorencio** [rudderlabs/rudder-server#4187]
**Describe the bug**
We recently upgraded from rudder-server 1.7.3 to 1.17.2, and I noticed that rudder-server's and PostgreSQL's CPU increased by an order of magnitude. I bisected the releases and saw that 1.17.0-rc1 is the first release to have this issue, so 1.16.3 doesn't have this performance regression. I can replicate this locally via docker and deployed to our k8s environment.

rudder-server 1.17.2 docker cpu, zero traffic:
<img width="575" alt="Screenshot 2023-12-04 at 3 53 47 PM" src="https://github.com/rudderlabs/rudder-server/assets/278027/f0b0b5eb-614f-4e6f-a8d0-1e78177da105">

rudder-server 1.16.3 docker cpu, zero traffic:
<img width="582" alt="Screenshot 2023-12-04 at 3 51 47 PM" src="https://github.com/rudderlabs/rudder-server/assets/278027/56ca3bb1-e1c8-4f73-baf9-0ebc8b5332a7">

rudder-server 1.7.3 -> 1.17.2 -> 1.16.3 container cpu on our low traffic staging environment (x86_64 GKE cluster):
<img width="769" alt="Screenshot 2023-12-04 at 3 55 31 PM" src="https://github.com/rudderlabs/rudder-server/assets/278027/df0b586e-0740-470a-9e40-f53f64b57811">

**Steps to reproduce the bug**

My local docker for rudder-server has the following environment variables set:
```
- JOBS_DB_HOST=(...)
- JOBS_DB_USER=(...)
- JOBS_DB_PORT=5432
- JOBS_DB_DB_NAME=(...)
- JOBS_DB_PASSWORD=(...)
- STATSD_SERVER_URL=statsd-exporter:9125
- DEST_TRANSFORM_URL=http://d-transformer:9090
- RSERVER_BACKEND_CONFIG_CONFIG_FROM_FILE=true
- RSERVER_BACKEND_CONFIG_CONFIG_JSONPATH=/etc/rudderstack/workspaceConfig.json
- RSERVER_LOGGER_MODULE_LEVELS=backend-config=WARN
- BUGSNAG_KEY=(...)
```
The workspaceConfig.json file is downloaded from rudderstack directly, and represents our workspace config on app.rudderstack.com.

If relevant, our source types are:
- Java
- JavaScript
- iOS

and destinations are:
- Webhook
- BigQuery
- GCP PubSub
- Heap
- Facebook Pixel
- Kafka

**Expected behavior**
Minimal CPU usage if there is no traffic.

**Comment by cwarren79** [comment]
I came here to report the same thing.

Using tcpdump on the host, it looks like rudder-server is looping with a high rate on this query
```
    172.17.0.5.51178 > 172.17.0.2.5432: Flags [P.], cksum 0x5959 (incorrect -> 0x5dbb), seq 439712:439977, ack 506484, win 1414, options [nop,nop,TS val 712569135 ecr 603348271], length 265
E..=..@................8....S..7....YY.....
*x./#.]/P.....
		SELECT

		id,
		source_id,
		destination_id,
		status,
		created_at,
		updated_at,
		tablename,
		error,
		async_job_type,
		metadata,
		attempt,
		workspace_id

		FROM
			wh_async_jobs
		WHERE
			status = $1 OR status = $2
		LIMIT $3;
	...D....S.S....
```

I don't see this happening on a server running `1.16.3`

**Comment by cisse21** [comment]
Thanks for raising the issue. We identified the root cause and are working on a fix for that

**Comment by achettyiitr** [comment]
https://github.com/rudderlabs/rudder-server/pull/4198

**Comment by achettyiitr** [comment]
Fixed is in release [v1.17.3](https://github.com/rudderlabs/rudder-server/releases/tag/v1.17.3)

**Comment by jflorencio** [comment]
Can confirm that the new release fixes the problem! Thanks for the quick fix 😄
